library/CMakeLists.txt

```diff
@@ -219,11 +219,11 @@ if(WIN32)
     set(libs ${libs} ws2_32 bcrypt)
 endif(WIN32)
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```

Comment:
Shouldn't it be:

    if(CMAKE_C_COMPILER_ID MATCHES "AppleClang" OR ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")

Reply:
No, because we are explicitly checking for apple clang. The fix is for the error we get when using regular clang *on macOS* (e.g. `brew install llvm`). In that case we do not want to add the flag, which `llvm-ranlib` does not recognize.